src/SDK/SmartDetectors.Extensions/Extensions/AlertExtensions.cs

```diff
@@ -122,11 +112,34 @@ public static ContractsAlert CreateContractsAlert(this Alert alert, SmartDetecto
                 AlertProperties = alertProperties,
                 RawProperties = rawProperties,
                 QueryRunInfo = queryRunInfo,
-                SignalType = signalType
+                SignalType = signalType,
+                AutomaticResolutionParameters = alert.AutomaticResolutionParameters?.CreateContractsAutomaticResolutionParameters()
             };
             #pragma warning restore CS0612 // Type or member is obsolete; Task to remove obsolete code #1312924
         }
 
+        /// <summary>
+        /// Extracts the predicate properties from the alert. Predicate properties are properties
+        /// in the alert's object which are marked by <see cref="PredicatePropertyAttribute"/>.
+        /// </summary>
+        /// <param name="alert">The alert to extract the predicates from.</param>
+        /// <returns>A dictionary mapping each predicate property name to its value.</returns>
+        public static Dictionary<string, object> ExtractPredicates(this Alert alert)
+        {
+            if (alert == null)
```

Comment:
This is an extension method, alert cannot be null

Reply:
of course it can be

---
In reply to: [234993635](https://github.com/Azure/SmartDetectorsMonitoringAppliance/pull/27#discussion_r234993635) [](ancestors = 234993635)